google_guest_agent/events/metadata/metadata.go

```diff
@@ -55,8 +55,13 @@ func (mp *Watcher) ID() string {
 	return WatcherID
 }
 
+// Events returns an slice with all implemented events.
+func (mp *Watcher) Events() []string {
+	return []string{LongpollEvent}
+}
+
 // Run listens to metadata changes and report back the event.
-func (mp *Watcher) Run(ctx context.Context) (bool, string, interface{}, error) {
+func (mp *Watcher) Run(ctx context.Context, evType string) (bool, interface{}, error) {
```

Comment:
I know right now there's only one event type but do we also want to refactor `Run()` to execute based on the received `evType`? Or that's for future whenever watcher introduces new event type?

Reply:
Do you mean in the Watcher implementations?

Reply:
Oh yeah! you are quoting the watcher implementation...

Reply:
Yep, for sure, so if we want to support multiple events the Run() implementation will decide what event to handle depending on the receiving evType... for metadata for example we can add a "first-metadata-ready" and handle it differently ....

Reply:
Right, I meant do you intend to refactor it in this PR or later when we introduce more events? 

Basically have a switch case or something and `Run()` executes only on `LongpollEvent evType` otherwise throw an error or no-op.